src/dev/run_extract_default_translations.js

```diff
@@ -0,0 +1,343 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import fs from 'fs';
+import glob from 'glob';
+import { resolve as resolvePath } from 'path';
+import { jsdom } from 'jsdom';
+import { parse } from '@babel/parser';
+import traverse from 'babel-traverse';
+import * as babelTypes from 'babel-types';
+import { promisify } from 'util';
+
+import { run } from './run';
+
+const defaultMessageKey = 'defaultMessage';
+const angularExpressionRegEx = /(\{)\{+([\s\S]*?)(\})\}+/g;
+
+const readFile = promisify(fs.readFile);
+const writeFile = promisify(fs.writeFile);
+const makeDir = promisify(fs.mkdir);
+const globProm = promisify(glob);
+
+run(async () => {
+  for (const inputPath of process.argv.slice(2)) {
+    await extractDefaultTranslations(inputPath);
+  }
+});
+
+async function isPathExists(path) {
+  await promisify(fs.access)(path, fs.constants.F_OK);
+}
+
+function addMessageToMap(targetMap, key, value) {
+  if (targetMap.has(key) && targetMap.get(key) !== value) {
+    throw new Error(`Default messages are different for id "${key}"`);
+  }
+  targetMap.set(key, value);
+}
+
+function parseConditionalOperatorAST(subTree) {
```

Comment:
__suggestion:__ would be great if you can add more comments and clarifications for these parsing methods. It's hard to understand what we're doing here and why.